packages/dashboard/src/components/DashboardFilters/DashboardFiltersEditor/components/FilterEditor.tsx

```diff
@@ -146,80 +166,24 @@ const FilterEditor: React.FC<FilterEditorProps> = ({
     return (
-      <>
-        {!isSubgroup && (
-          <TextField
-            label='API Endpoint: '
-            value={filter.apiFilter?.apiEndpoint}
-            updateField={(_section, _subSection, _key, value) => updateAPIFilter('apiEndpoint', value)}
-            tooltip={
-              <>
-                {isNestedDropDown && (
-                  <Tooltip style={{ textTransform: 'none' }}>
-                    <Tooltip.Target>
-                      <Icon display='question' style={{ marginLeft: '0.5rem' }} />
-                    </Tooltip.Target>
-                    <Tooltip.Content>
-                      <p>Your API Endpoint should return both value selector values.</p>
-                    </Tooltip.Content>
-                  </Tooltip>
-                )}
-              </>
-            }
-          />
-        )}
-
-        <div className={isNestedDropDown ? 'border border-dark p-1 my-1' : ''}>
-          <TextField
-            label={`${isSubgroup ? 'Subgroup ' : ''}Value Selector:`}
-            value={filter?.apiFilter?.[valueSelector] || ''}
-            updateField={(_section, _subSection, _key, value) => updateAPIFilter(valueSelector, value)}
-            tooltip={
-              <>
-                <Tooltip style={{ textTransform: 'none' }}>
-                  <Tooltip.Target>
-                    <Icon display='question' style={{ marginLeft: '0.5rem' }} />
-                  </Tooltip.Target>
-                  <Tooltip.Content>
-                    <p>Value to use in the html option element</p>
-                  </Tooltip.Content>
-                </Tooltip>
-                {` * Required`}
-              </>
-            }
-          />
-
-          <TextField
-            label={`${isSubgroup ? 'Subgroup ' : ''}Display Text Selector:`}
-            value={filter?.apiFilter?.[textSelector] || ''}
-            updateField={(_section, _subSection, _key, value) => updateAPIFilter(textSelector, value)}
-            tooltip={
-              <>
-                <Tooltip style={{ textTransform: 'none' }}>
-                  <Tooltip.Target>
-                    <Icon display='question' style={{ marginLeft: '0.5rem' }} />
-                  </Tooltip.Target>
-                  <Tooltip.Content>
-                    <p>Text to use in the html option element. If none is applied value selector will be used.</p>
-                  </Tooltip.Content>
-                </Tooltip>
-                {` * Optional`}
-              </>
-            }
-          />
-        </div>
-      </>
+      <Modal>
+        <Modal.Content>
+          <APIModal filter={filter} isNestedDropdown={isNestedDropdown} updateAPIFilter={updateAPIFilter} />
+        </Modal.Content>
+      </Modal>
     )
   }
+  const { overlay } = useGlobalContext()
+
+  const handleEditAPIValues = (filter, isNestedDropdown, updateAPIFilter) => {
+    {
+      overlay.actions.openOverlay(EditAPIValues(filter, isNestedDropdown, updateAPIFilter))

```

Comment:
just add this in directly:

<Modal>
        <Modal.Content>
          <APIModal filter={filter} isNestedDropdown={isNestedDropdown} updateAPIFilter={updateAPIFilter} />
        </Modal.Content>
      </Modal>